Utils/FaviconFetcher.swift

```diff
@@ -13,108 +16,118 @@ class FaviconFetcherErrorType: ErrorType {
+        configuration.timeoutIntervalForRequest = 5
+        configuration.HTTPAdditionalHeaders = defaultHeaders
+
+        return Alamofire.Manager(configuration: configuration)
+    }()
+
+    private func fetchDataForUrl(url: NSURL) -> Deferred<Result<NSData>> {
+        let deferred = Deferred<Result<NSData>>()
+        alamofire.request(.GET, url).response { (request, response, data, error) in
+            if let error = error {
+                deferred.fill(Result(failure: FaviconFetcherErrorType(description: error.description)))
+            } else {
+                let loc = response?.URL
+                deferred.fill(Result(success: data as! NSData))
+            }
         }
+        return deferred
     }
 
     // Loads and parses an html document and tries to find any known favicon-type tags for the page
-    private func loadFromDoc() {
+    private func parseHTMLForFavicons(url: NSURL) -> Deferred<Result<[Favicon]>> {
         var err: NSError?
 
-        if let data = NSData(contentsOfURL: siteUrl),
-           let element = RXMLElement(fromHTMLData: data) {
-            element.iterate("head.meta") { meta in
-                if let refresh = meta.attribute("http-equiv"),
-                   let content = meta.attribute("content"),
-                   let index = content.rangeOfString("URL="),
-                   let url = NSURL(string: content.substringFromIndex(advance(index.startIndex,4))) {
-                    if refresh == "Refresh" {
-                        self.siteUrl = url
-                        self.loadFromDoc()
-                        return
+        return fetchDataForUrl(url).bind({ result -> Deferred<Result<[Favicon]>> in
+            var icons = [Favicon]()
+
+            if let data = result.successValue,
+               let element = RXMLElement(fromHTMLData: data) {
+                var reloadUrl: NSURL? = nil
+                element.iterate("head.meta") { meta in
+                    if let refresh = meta.attribute("http-equiv") where refresh == "Refresh",
+                        let content = meta.attribute("content"),
+                        let index = content.rangeOfString("URL="),
+                        let url = NSURL(string: content.substringFromIndex(advance(index.startIndex,4))) {
+                            reloadUrl = url
                     }
                 }
-            }
 
-            element.iterate("head.link") { link in
-                if var rel = link.attribute("rel") where (rel == "shortcut icon" || rel == "icon" || rel == "apple-touch-icon"),
-                    var href = link.attribute("href"),
-                    var url = NSURL(string: href, relativeToURL: self.siteUrl) {
-                        let icon = Favicon(url: url.absoluteString!, date: NSDate(), type: IconType.Icon)
-                        self._favicons.append(icon)
+                if let url = reloadUrl {
+                    return self.loadFromLinks(url)
```

Comment:
Note that this method doesn't exist in this PR, but I see it's in the other PR. Just make sure not to land this PR by itself.